more native filechooser dialogs

Mark-64 opened the pull request:
See posts below

---
This PR concludes the migration to Gtk native file choosers for image I/O operations.
Also I changed some strings to be more similar to the older dialogs

What is done now is:

- Import, Copy & Import
- Export to disk, latex, gallery
- Copy / Move
- Search Filmroll

so basically all what concerns images I/O is migrated. What is left on Gtk non native dialogs is I/O about:

- acceletarors
- presets
- styles
- preferences
- history
- tagging
- geotagging
- 3D luts

I am not sure the above is worth to migrate

johnny-bit commented:
Using native dialogs everywhere brings us closer to the universal os experience for everything plus a solid way to handle sandboxed envs like flatpack, so migrating all is good idea imo. As a bonus it's a little stone forward if we'd ever want GTK4.

Mark-64 commented:
Ok, I will continue here then.

Mark-64 commented:
Ok, now with this PR the migration to native file choosers is complete, with two exceptions, which can't be simply migrated beacuse of special features not supported by the native file chooser. 

1. geotagging, because it has a custom preview
2. darktable-chart, because it has an extra widget

Also, now that the native choosers have many different tasks, in Windows they can't remember many different places, so I modified the conf <-> filechooser functions to workaround the issue with `gtk_file_chooser_get_current_folder()`, instead of simply returning.
Finally, fixed an issue with import presets always giving a false error message.

Ready for review

TurboGit reviewed (approved):
Works for me, thanks !

johnny-bit commented:
awesome work! thanks!
